packages/animations/test/page_transition_switcher_test.dart

```diff
@@ -0,0 +1,622 @@
+    final UniqueKey statefulOne = UniqueKey();
+    final UniqueKey statefulTwo = UniqueKey();
+    final UniqueKey statefulThree = UniqueKey();
+
+    StatefulTestState.generation = 0;
+
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: StatefulTest(key: statefulOne),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    Map<Key, double> _primaryAnimation =
+        _getPrimaryAnimation(<Key>[statefulOne], tester);
+    Map<Key, double> _secondaryAnimation =
+        _getSecondaryAnimation(<Key>[statefulOne], tester);
+    expect(_primaryAnimation[statefulOne], equals(1.0));
+    expect(_secondaryAnimation[statefulOne], equals(0.0));
+    expect(StatefulTestState.generation, equals(1));
+
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: StatefulTest(key: statefulTwo),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 50));
+    expect(find.byType(FadeTransition), findsNWidgets(2));
+    _primaryAnimation =
+        _getPrimaryAnimation(<Key>[statefulOne, statefulTwo], tester);
+    _secondaryAnimation =
+        _getSecondaryAnimation(<Key>[statefulOne, statefulTwo], tester);
+    expect(_primaryAnimation[statefulTwo], equals(0.5));
+    expect(_secondaryAnimation[statefulTwo], equals(0.0));
+    expect(StatefulTestState.generation, equals(2));
+
+    await tester.pumpWidget(
+      PageTransitionSwitcher(
+        duration: const Duration(milliseconds: 100),
+        child: StatefulTest(key: statefulThree),
+        transitionBuilder: _transitionBuilder,
+      ),
+    );
+
+    await tester.pump(const Duration(milliseconds: 10));
+    expect(StatefulTestState.generation, equals(3));
+    await tester.pumpAndSettle();
+    expect(StatefulTestState.generation, equals(3));
+  });
+
+  testWidgets('updates widgets without animating if they are isomorphic.',
+      (WidgetTester tester) async {
+    Future<void> pumpChild(Widget child) async {
+      return tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.rtl,
+          child: PageTransitionSwitcher(
+            duration: const Duration(milliseconds: 100),
+            child: child,
+            transitionBuilder: _transitionBuilder,
+          ),
+        ),
+      );
+    }
+
+    await pumpChild(const Text('1'));
+    await tester.pump(const Duration(milliseconds: 10));
+    FadeTransition fade = tester.widget(find.byType(FadeTransition).first);
+    ScaleTransition scale = tester.widget(find.byType(ScaleTransition).first);
+    expect(fade.opacity.value, equals(1.0));
+    expect(scale.scale.value, equals(1.0));
+    expect(find.text('1'), findsOneWidget);
+    expect(find.text('2'), findsNothing);
+    await pumpChild(const Text('2'));
+    fade = tester.widget(find.byType(FadeTransition).first);
```

Comment:
Shouldn't there only be one widget, since it's not animating and simply updating the value?